sdk/go/common/util/gitutil/git_test.go

```diff
@@ -342,3 +381,95 @@ func TestTryGetVCSInfoFromSSHRemote(t *testing.T) {
 		assert.Equal(t, test.WantVCSInfo, got)
 	}
 }
+
+func TestParseAuthURL(t *testing.T) {
+	t.Parallel()
+
+	//nolint: gosec
+	generateSSHKey := func(t *testing.T, passphrase string) string {
+		// TODO(go1.21): clear(sshKeys)
+		for m := range sshKeys {
+			delete(sshKeys, m)
+		}
+
+		r := rand.New(rand.NewSource(0))
+		key, err := rsa.GenerateKey(r, 256)
+		require.NoError(t, err)
+
+		block := &pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(key),
+		}
+
+		if passphrase != "" {
+			//nolint: staticcheck
+			block, err = x509.EncryptPEMBlock(r, block.Type, block.Bytes, []byte(passphrase), x509.PEMCipherAES256)
+			require.NoError(t, err)
+		}
+
+		path := path.Join(t.TempDir(), "test-key")
```

Comment:
Should this be filepath.Join?
```suggestion
		path := filepath.Join(t.TempDir(), "test-key")
```

path.Join operates on `/`-separated paths. filepath.Join uses OS-specific file spearators.